src/backend/distributed/shared_library_init.c

```diff
@@ -327,6 +361,18 @@ RegisterCitusConfigVariables(void)
 		NULL, NULL, NULL);
 
 	DefineCustomBoolVariable(
+		"citus.log_distributed_deadlock_detection",
+		gettext_noop("Log distributed deadlock detection related processing in "
+					 "the server log"),
+		NULL,
+		&LogDistributedDeadlockDetection,
+		false,
+		PGC_USERSET,
```

Comment:
I don't think `USERSET` is the right category, setting it in a session won't have an effect. I think it should be `PGC_SIGHUP`, no?